Assets/MixedRealityToolkit/Inspectors/Profiles/BaseMixedRealityProfileInspector.cs

```diff
@@ -78,8 +139,45 @@ private static bool RenderProfileInternal(SerializedProperty property, GUIConten
                 }
             }
 
+            // Begin the horizontal group
             EditorGUILayout.BeginHorizontal();
-            RenderProfileField(property, guiContent, GetProfileTypesForService(serviceType));
+
+            // Draw the foldout - this contains the property name so it's clickable
+            bool showFoldout = false;
+            if (oldObject != null)
+            {
+                string showFoldoutKey = GetSubProfileDropdownKey(property);
+                showFoldout = SessionState.GetBool(showFoldoutKey, false);
+                showFoldout = EditorGUILayout.Foldout(showFoldout, property.displayName, true);
+                SessionState.SetBool(showFoldoutKey, showFoldout);
+            }
+
+            // Find the profile type so we can limit the available object field options
+            Type profileType = null;
+            if (serviceType != null)
+            {
+                // If profileTypes has a count greater than one, then it won't be possible to use
+                // EditorGUILayout.ObjectField to restrict the set of profiles to a single type - in this
+                // case all profiles of BaseMixedRealityProfile will be visible in the picker.
```

Comment:
This comment I think was copy pasted from the remarks but it's a bit stale (i.e. it references a types count which doesn't exist within this scope)

Reply:
And also I think that this still leads to an issue where if there are 2+ types it will always only return the first type (is this just something that will never happen?). Basically the comments no longer match the code (i.e. it used to be that 0 == show everything, 1== show only that type, 2 == show everything) but now it's (0 == show everything, 1 == show only that type, 2 == show a random one based on however the underlying collection enumeration works)

---
In reply to: [279475209](https://github.com/Microsoft/MixedRealityToolkit-Unity/pull/3773#discussion_r279475209) [](ancestors = 279475209)